File: Serverside/aspnet-core/src/Homies.RealEstate.Core/Server/DanhMuc.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Abp.Domain.Entities.Auditing;
using Abp.Domain.Entities;

namespace Homies.RealEstate.Server
{
    [Table("DanhMucs")]
    public class DanhMuc : Entity
    {

        [Required]
        [StringLength(DanhMucConsts.MaxTenDanhMucLength, MinimumLength = DanhMucConsts.MinTenDanhMucLength)]
        public virtual string TenDanhMuc { get; set; }

        public virtual string Tag { get; set; }

        [Required]
        [StringLength(DanhMucConsts.MaxTrangThaiLength, MinimumLength = DanhMucConsts.MinTrangThaiLength)]
        public virtual string TrangThai { get; set; }

        public virtual int? DanhMucCha { get; set; }

    }
}

File: Clientside/flutter/lib/models/danhMuc/danhMuc.dart
class DanhMuc {
  int id;
  String tenDanhMuc;
  String tag;
  String trangThai;
  int danhMucCha;

  DanhMuc({
    this.id,
    this.tenDanhMuc,
    this.tag,
    this.trangThai,
    this.danhMucCha,
  });

  factory DanhMuc.fromMap(Map<String, dynamic> json) => DanhMuc(
    id: json["id"],
    tenDanhMuc: json["tenDanhMuc"],
    tag: json["tag"],
    trangThai: json["trangThai"],
    danhMucCha: json["danhMucCha"],
  );
}